packages/ui/src/components/GalleryPost/GalleryPost.tsx

```diff
@@ -64,11 +70,13 @@ export default function GalleryPost({
   const previewType =
     isImage || isLinkedImage
       ? 'image'
-      : isText && !isLinkedImage && !isRefInText
+      : isText && !isLinkedImage && !isRefInText && !isLink
         ? 'text'
         : isReference || isRefInText
           ? 'reference'
-          : 'unsupported';
+          : isLink
+            ? 'link'
+            : 'unsupported';
```

Comment:
Would probably be good to rewrite this without the ternaries now that it's gotten so complex.